iothub_client/tests/common_e2e/iothubclient_common_e2e.c

```diff
@@ -342,50 +368,101 @@ static void ReceiveUserContext_Destroy(EXPECTED_RECEIVE_DATA* data)
         {
             ASSERT_FAIL("unable to Lock_Init");
+            free(result);
+            result = NULL;
         }
         else
         {
-            char temp[1000];
-            char* tempString;
-            time_t t = time(NULL);
-            size_t string_length;
-            const char* data_fmt;
+            result->expectedString = string;
+            result->wasFound = false;
+            result->dataWasRecv = false;
+            result->result = IOTHUB_CLIENT_CONFIRMATION_ERROR;
+        }
+    }
 
-            if (g_e2e_test_options.use_special_chars)
-            {
-                data_fmt = TEST_EVENT_DATA_FMT_SPECIAL_CHAR;
-            }
-            else
-            {
-                data_fmt = TEST_EVENT_DATA_FMT;
-            }
+    return result;
+}
 
-            string_length = sprintf(temp, data_fmt, ctime(&t), g_iotHubTestId);
-            if ((tempString = (char*)malloc(string_length + 1)) == NULL)
-            {
-                Lock_Deinit(result->lock);
-                free(result);
-                result = NULL;
-            }
-            else
-            {
-                strcpy(tempString, temp);
-                result->expectedString = tempString;
-                result->wasFound = false;
-                result->dataWasRecv = false;
-                result->result = IOTHUB_CLIENT_CONFIRMATION_ERROR;
-            }
+static EXPECTED_SEND_DATA* EventData_Create(void)
+{
+    EXPECTED_SEND_DATA* result;
+    char temp[1000];
+    char* tempString;
+    time_t t = time(NULL);
+    size_t string_length;
+    const char* data_fmt;
+
+    if (g_e2e_test_options.use_special_chars)
+    {
+        data_fmt = TEST_EVENT_DATA_FMT_SPECIAL_CHAR;
+    }
+    else
+    {
+        data_fmt = TEST_EVENT_DATA_FMT;
+    }
+
+    string_length = sprintf(temp, data_fmt, ctime(&t), g_iotHubTestId++);
+    if ((tempString = (char*)malloc(string_length + 1)) == NULL)
+    {
+        result = NULL;
+    }
+    else
+    {
+        (void)strcpy(tempString, temp);
+        result = EventData_Create_With_String(tempString);
+
+        if (result == NULL)
+        {
+            free(tempString);
         }
     }
+
+    return result;
+}
+
+static EXPECTED_SEND_DATA* EventData_Create_With_Custom_Size(size_t messageSize)
+{
+    EXPECTED_SEND_DATA* result;
+    char* message = malloc(sizeof(char) * messageSize);
+
+    if (message == NULL)
+    {
+        LogError("Failed to allocate EventData message");
+        result = NULL;
+    }
+    else
+    {
+        char idString[10];
+        size_t idStringLength = sprintf(idString, "%d", (int)g_iotHubTestId++);
+
+        (void)memset(message, (int)'a', messageSize - 1);
+        (void)memcpy(message + 1, idString, idStringLength);
+        message[messageSize - 1] = '\0';
+
+        result = EventData_Create_With_String(message);
+
+        if (result == NULL)
+        {
+            LogError("Failed allocating EventData");
```

Comment:
Similar to above I think this should just be `ASSERT_NOT_NULL(result)`;;  

There's some other places that have been modified I think can follow the ASSERT pattern which I'm not going to call out individually.

Reply:
Done.